compiling for windows

Hi,

Trying to compile version 0.5.9 for x86_64-pc-windows-gnu fails.

error[E0425]: cannot find function `sysconf` in crate `libc`
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/sys-info-0.5.9/lib.rs:396:34
    |
396 |             let buf_size = libc::sysconf(libc::_SC_HOST_NAME_MAX) as usize;
    |                                  ^^^^^^^ not found in `libc`

error[E0425]: cannot find value `_SC_HOST_NAME_MAX` in crate `libc`
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/sys-info-0.5.9/lib.rs:396:48
    |
396 |             let buf_size = libc::sysconf(libc::_SC_HOST_NAME_MAX) as usize;
    |                                                ^^^^^^^^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find function `gethostname` in crate `libc`
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/sys-info-0.5.9/lib.rs:398:22
    |
398 |             if libc::gethostname(buf.as_mut_ptr() as *mut i8, buf_size) < 0 {
    |                      ^^^^^^^^^^^ help: a function with a similar name exists: `getsockname`

error: aborting due to 3 previous errors



Version 0.5.8 works fine.

Amir

--- a/lib.rs
+++ b/lib.rs
@@ -390,8 +390,8 @@ pub fn disk_info() -> Result<DiskInfo, Error> {
 }
 
 /// Get hostname.
+#[cfg(target_family = "unix")]
 pub fn hostname() -> Result<String, Error> {
-    if cfg!(unix) {
         unsafe {
             let buf_size = libc::sysconf(libc::_SC_HOST_NAME_MAX) as usize;
             let mut buf = Vec::<u8>::with_capacity(buf_size + 1);
@@ -402,15 +402,15 @@ pub fn hostname() -> Result<String, Error> {
             buf.set_len(hostname_len);
             Ok(ffi::CString::new(buf).unwrap().into_string().unwrap())
         }
-    } else if cfg!(windows) {
-        use std::process::Command;
-        Command::new("hostname")
-            .output()
-            .map_err(Error::ExecFailed)
-            .map(|output| String::from_utf8(output.stdout).unwrap().trim().to_string())
-    } else {
-        Err(Error::UnsupportedSystem)
-    }
+}
+
+#[cfg(target_family = "windows")]
+pub fn hostname() -> Result<String, Error> {
+    use std::process::Command;
+    Command::new("hostname")
+        .output()
+        .map_err(Error::ExecFailed)
+        .map(|output| String::from_utf8(output.stdout).unwrap().trim().to_string())
 }
 
 /// Get system boottime

